### PR TITLE
[UOT-121082] Update GC ML API expandTerms endpoint to support JBook QE model

### PR DESCRIPTION
**Pull request by mishoe**
**Jira Ticket:** https://jira.di2e.net/browse/UOT-121082
**Description**: A new QE index has been created for the JBOOK search tool. This code update is to modify the existing GC ML API `expandTerms` endpoint to support utilizing the jbook QE index if a request comes in from the jbook search tool

The endpoint now checks the payload that is posted to it to determine what `qe_model` to use. By default, if not passed in a value for `qe_model`, the endpoint will leverage the existing query expansion model (see below updated logic for endpoint, defaulting to GC QE if `qe_model`!=`jbook`)
Screen Shot 2021-11-10 at 11 59 04 AM

### CODE REVIEW

File: gamechangerml/api/fastapi/routers/search.py [hunk 1]
```diff
@@ -152,10 +152,11 @@ async def post_expand_query_terms(termsList: dict, response: Response) -> dict:
     # logger.info("[{}] expanded: {}".format(user, termsList))
 
     logger.info(f"Expanding: {termsList}")
+    query_expander = MODELS.query_expander if termsList.get("qe_model","gc_core")!="jbook" else MODELS.query_expander_jbook
```
Comment (mishoe) [inline gamechangerml/api/fastapi/routers/search.py:155]: Reviewers: Do we have a preference over what the `key:value` param should be called for delineating whether the endpoint utilizes the JBook QEXP model vs the O.G. GC model when performing QEXP?

File: gamechangerml/api/utils/pathselect.py [hunk 2]
```diff
@@ -12,7 +12,7 @@ def get_model_paths():
         qexp_names = [
             f
             for f in os.listdir(Config.LOCAL_PACKAGED_MODELS_DIR)
-            if ("qexp_" in f) and ("tar" not in f)
+            if ("qexp_" in f) and (all(substr not in f for substr in ["tar","jbook"]))
```
Comment (mishoe) [inline gamechangerml/api/utils/pathselect.py:15]: the GC core model folder criteria is now such that it contains "qexp_" but doesn't contain either "tar" or "jbook"

File: gamechangerml/setup_env.sh [hunk 3]
```diff
@@ -20,6 +20,7 @@ function setup_prod() {
     export S3_TRANS_MODEL_PATH="${S3_TRANS_MODEL_PATH:-s3://advana-data-zone/bronze/gamechanger/models/transformers/v5/transformers.tar.gz}"
     export S3_SENT_INDEX_PATH="${S3_SENT_INDEX_PATH:-s3://advana-data-zone/bronze/gamechanger/models/sentence_index/v7/sent_index_20210715.tar.gz}"
     export S3_QEXP_PATH="${S3_QEXP_PATH:-s3://advana-data-zone/bronze/gamechanger/models/qexp_model/v3/qexp_20201217.tar.gz}"
+    export S3_QEXP_JBOOK_PATH="${S3_QEXP_JBOOK_PATH:-s3://advana-data-zone/bronze/gamechanger/models/jbook_qexp_model/v1/jbook_qexp_20211029.tar.gz}"
```
Comment (mishoe) [inline gamechangerml/setup_env.sh:23]: Reviewers (or GC ML team): we've created a `models/jbook_qexp_model` folder to stage the jbook qexp model into - any preference towards having the model in a different folder?

File: gamechangerml/src/search/query_expansion/qe.py [hunk 4]
```diff
@@ -119,8 +119,8 @@ def _similar_to(self, query_str, topn, threshold):
             logger.debug("{}".format(query_str.lower().strip()))
             logger.debug(" cos sim {}".format(final_cos))
             logger.debug("expanded {}".format(expanded))
-            logger.debug("   vocab {}".format(
-                [self._vocab[idx] for idx in v_idx]))
+            # logger.debug("   vocab {}".format(
```
Comment (mishoe) [inline gamechangerml/src/search/query_expansion/qe.py:122]: commented this debug, as it is overly verbose.

**Comment by mishoe** [comment]
Hey @rha930 & @cWren0110 - this PR is ready for review when you have availability 👍

**Review by rha930 — APPROVED** [review]

**Pull request merged**